lib/samples/apply_mosaic_rule_to_rasters/apply_mosaic_rule_to_rasters.dart

```diff
@@ -0,0 +1,200 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:flutter/material.dart';
+
+enum MosaicMethodEnum {
+  objectID('Object ID', 'Orders rasters based on the order (ObjectID).'),
+  northwest(
+    'North West',
+    'Orders rasters based on the distance between each raster center and the northwest point.',
+  ),
+  center(
+    'Center',
+    'Orders rasters based on the distance between each raster center and the view center.',
+  ),
+  attribute(
+    'By Attribute',
+    'Orders rasters based on the absolute distance between their values of an attribute and a base value.',
+  ),
+  lockRaster(
+    'Lock Raster',
+    'Displays only the selected rasters specified in [MosaicRule.lockRasterIds].',
+  );
+
+  const MosaicMethodEnum(this.value, this.description);
+  final String value;
+  final String description;
+}
+
+class ApplyMosaicRuleToRasters extends StatefulWidget {
+  const ApplyMosaicRuleToRasters({super.key});
+
+  @override
+  State<ApplyMosaicRuleToRasters> createState() =>
+      _ApplyMosaicRuleToRastersState();
+}
+
+class _ApplyMosaicRuleToRastersState extends State<ApplyMosaicRuleToRasters>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // If the mosaic options should be shown.
+  var _showMosaicOptions = false;
+  // Current selected mosaic method.
+  var _selectedMosaicMethod = MosaicMethodEnum.objectID;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+      bottomSheet: _buildBottomSheet(),
+    );
+  }
+
+  Future<void> onMapViewReady() async {
+    // Create a map with a topographic basemap style.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    _mapViewController.arcGISMap = map;
+    //Create a Raster with the provided URL
```

Comment:
Some of the comments in this method should have a space after the `//` and a period at the end.